cmd/upgrade/upgrade.go

```
@@ -0,0 +1,380 @@
+		"@echo off\r\n"+
+			":loop\r\n"+
+			"timeout /t 1 /nobreak >nul\r\n"+
+			"del \"%s\" 2>nul\r\n"+
+			"if exist \"%s\" goto loop\r\n"+
+			"move \"%s\" \"%s\"\r\n"+
+			"del \"%%~f0\"\r\n",
+		execPath, execPath, newPath, execPath,
+	)
+
+	if err := os.WriteFile(scriptPath, []byte(script), 0o644); err != nil {
+		os.Remove(newPath)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)